acct/update-email

```diff
@@ -34,111 +26,104 @@ there is a problem with your account.
 
+    mail_attr = list(filter(
+        lambda attr: attr.startswith('{}: '.format(LDAP_MAIL_ATTR)),
+        output
+    ))
 
-def set_email(conn, dn, old_email, new_email):
-    """Updates email attribute in LDAP."""
-    old = {MAIL_ATTR: '*'}
-    new = {MAIL_ATTR: new_email}
+    if mail_attr:
+        # Strip the '{LDAP_MAIL_ATTR}: ' from the beginning of the string
+        return mail_attr[0][len(LDAP_MAIL_ATTR) + 2:].strip()
 
-    # we either need to provide old as {} or {'mail': '*'}; it will do
-    # nothing if we use the wrong one
-    #
-    # we know which one should work, but if it fails (e.g. due to a race
-    # condition), try the other one
-    def attempt_modify(modify):
-        ldif = ldap.modlist.modifyModlist(old if modify else {}, new)
-        conn.modify_s(dn, ldif)
 
-    modify = old_email is not None
+def set_email(username, new_email):
+    """Updates email attribute in LDAP."""
 
     try:
-        attempt_modify(modify)
-    except ldap.NO_SUCH_ATTRIBUTE:
-        attempt_modify(not modify)
-
-
-def get_dn():
-    return 'uid={},{}'.format(getpass.getuser(), LDAP_BASE)
+        modify_ldap_attributes(username, {LDAP_MAIL_ATTR: new_email})
+    except ValueError as e:
+        print(e)
+        sys.exit(1)
 
 
-if __name__ == '__main__':
+def main(argv=None):
     parser = argparse.ArgumentParser(description='Update OCF email')
-    parser.add_argument('--auto', action='store_true',
-                        help='only prompt if email is missing, suppress errors')
+    parser.add_argument(
+        '--auto',
+        action='store_true',
+        help='only prompt if email is missing, suppress errors',
+    )
     args = parser.parse_args()
 
-    conn = get_ldap()
+    # Check if the user has a valid kerberos ticket. If they don't, then they
+    # will have to enter their password in order to get a new ticket
+    has_kerberos = subprocess.run(('/usr/bin/klist', '-t')).returncode
 
-    if not conn:
-        if not args.auto:
-            print('Unable to load LDAP, try running kinit', file=sys.stderr)
-        sys.exit(1)
+    if has_kerberos != 0:
+        if args.auto:
+            # We can't ask for the user's password in auto mode, so we just exit
+            sys.exit(1)
+        else:
+            # The user did not already have a valid kerberos ticket, so we need
+            # to obtain a new one and ask for their password
+            subprocess.run('/usr/bin/kinit', check=True)
 
-    dn = get_dn()
-    email = get_email(conn, dn)
+    username = getpass.getuser()
+    email = get_email(username)
 
     if args.auto:
-        if valid_email(email):
+        if valid_non_ocf_email(email):
             sys.exit(0)
         else:
             print(AUTO_MESSAGE)
 
     print('Current Email Address: {}'.format(email or '(unset)'))
 
-    if valid_email(email) and input(Q_UPDATE_EMAIL) not in ('y', 'yes'):
+    if valid_non_ocf_email(email) and input(Q_UPDATE_EMAIL) not in ('y', 'yes'):
```

Comment:
```suggestion
    if valid_non_ocf_email(email) and input(Q_UPDATE_EMAIL) not in {'y', 'yes'}:
```

This is faster

Reply:
smh ethanhs is not google scale

Reply:
After running
```
python3 -m timeit "'y' not in {'y', 'yes'}"
python3 -m timeit "'y' not in ('y', 'yes')"
```

On `supernova`, I'm not actually convinced that either of these is faster than the other.

Reply:
Agreed, I took a look and `set` doesn't implement `tp_as_sequence`, while of course `tuple` does. So theoretically tuple should be faster (and testing indicates this).

Reply:
It actually depends. If you try running

```
python3 -m timeit "'n' not in {'y', 'yes'}"
python3 -m timeit "'n' not in ('y', 'yes')"
```

you might actually notice a difference.

In any case, no one is going to notice. Actually, a better argument might be semantic (think math notation).

Reply:
>It actually depends.

Yeah, that is a good point. I should have been more careful in benchmarking claims.

>a better argument might be semantic (think math notation).

I understand where you are coming from, however Python is not math, and I think it is generally more Pythonic to use a tuple here. Furthermore, looking at sourcegraph, most uses of `x in <sequence>`  use a tuple, so for the sake of consistency I would argue a tuple is the best choice, unless there is a reason not to use one.

Reply:
Addressing the two separate points that have been brought up here:

1. Performance of a two element tuple/set absolutely **do not matter** here. Unless this is taking some outrageous amount of time (like more than half a second or something), then performance-motivated changes should not be motivating changes on this.

2. A semantic argument makes a whole lot more sense, but given this is Python and @ethanhs's argument on this, I think sticking with a tuple makes sense and also is clearer to read since it's clear that it's a tuple and not a hash or something that also uses `{` and `}`.

Thoughts? Personally, I don't really think it matters, and the amount of discussion here is way more than is needed for such a small issue/change.

Reply:
I think my final opinion on the matter is the bikeshed should be blue.